### PR TITLE
Simplify Authentication classes and make them copyable/movable

### PR DESCRIPTION
**Pull request by jagerman**
I ran into an issue where I want to be able to assign a member Authentication variable from a provided Authentication value, but that is impossible because of the `const` private members.  Removing the `const` fixes this (and since they are already private doesn't change anything), but while I was here I also noticed some unnecessary related code, so simplified that as well.  See individual commit messages for details.

### CODE REVIEW

File: include/cpr/digest.h [hunk 1]
```diff
@@ -7,12 +7,7 @@ namespace cpr {
 
 class Digest : public Authentication {
   public:
-    Digest(std::string&& username, std::string&& password)
-            : Authentication{std::move(username), std::move(password)} {}
-
-    ~Digest() override = default;
-
-    const char* GetAuthString() const noexcept;
+    using Authentication::Authentication;
```
Comment (COM8) [inline include/cpr/digest.h:10]: I'm not a fan of this notation.

Could you please replace it with the appropriate overrides for constructors.

File: include/cpr/auth.h [hunk 2]
```diff
@@ -9,18 +9,14 @@ namespace cpr {
...
-            : username_(std::move(username)),
-              password_(std::move(password)), auth_string_{this->username_ + ":" +
-                                                           this->password_} {}
+    Authentication(const std::string& username, const std::string& password)
+            : auth_string_{username + ":" + password} {}
     virtual ~Authentication() = default;
 
     const char* GetAuthString() const noexcept;
 
   private:
```
Comment (COM8) [inline include/cpr/auth.h:18]: Could you change the visibility to protected for properties.

File: include/cpr/ntlm.h [hunk 3]
```diff
@@ -8,11 +8,7 @@ namespace cpr {
 
 class NTLM : public Authentication {
   public:
-    template <typename UserType, typename PassType>
-    NTLM(UserType&& username, PassType&& password)
-            : Authentication{std::move(username), std::move(password)} {}
-
-    const char* GetAuthString() const noexcept;
+    using Authentication::Authentication;
```
Comment (COM8) [inline include/cpr/ntlm.h:11]: Same problem like above - not a fan of this.

File: include/cpr/auth.h [hunk 4]
```diff
@@ -9,18 +9,14 @@ namespace cpr {
 
 class Authentication {
   public:
-    Authentication(std::string&& username, std::string&& password)
```
Comment (COM8) [inline include/cpr/auth.h:12]: Please also add rvalue constructors for `Authentication` to keep compatibility with older implementations.

File: include/cpr/digest.h (hunk 1 above)
Comment (jagerman) [reply include/cpr/digest.h:10]: I don't understand "not a fan" -- this is a bog standard C++11 inherited constructor.

File: include/cpr/auth.h (hunk 4 above)
Comment (jagerman) [reply include/cpr/auth.h:12]: Rvalues can be passed to const lvalue reference arguments, so this does not break compatibility.

File: include/cpr/auth.h (hunk 2 above)
Comment (jagerman) [reply include/cpr/auth.h]: Changed.

File: include/cpr/digest.h (hunk 1 above)
Comment (jagerman) [reply include/cpr/digest.h]: I put explicit constructors back in.

**Comment by jagerman** [comment]
@COM8 - one other possible change to be added here (checking first before I change it) is to make the `const char* GetAuthString() const noexcept;` virtual.  Without that, as it stands currently, there's no way for me to create a subclass that overrides it (because once it is cast to the base class the `auth.GetAuthString()` is not going to do a virtual method lookup).

File: include/cpr/auth.h (hunk 4 above)
Comment (COM8) [reply include/cpr/auth.h:12]: You are right.
Nevertheless, I'd like to keep the `rvalue` constructors for verbosity reasons.

**Comment by COM8** [comment]
> @COM8 - one other possible change to be added here (checking first before I change it) is to make the `GetAuthString()` method virtual. Without that, as it stands currently, there's no way for me to create a subclass that overrides it (because once it is cast to the base class and given to a `cpr` method the `auth.GetAuthString()` is not going to do a virtual method lookup).

@jagerman Thats a good idea. I'd say go for it :+1:

File: include/cpr/digest.h (hunk 1 above)
Comment (COM8) [reply include/cpr/digest.h]: Without the explicit constructors I have to open two files to find out, which constructors are provided.
So this is more work for me as a developer using `CPR`.

File: include/cpr/auth.h (hunk 4 above)
Comment (jagerman) [reply include/cpr/auth.h:12]: Done.

File: include/cpr/auth.h [hunk 5]
```diff
@@ -9,18 +9,19 @@ namespace cpr {
 
 class Authentication {
   public:
+    Authentication(const std::string& username, const std::string& password)
+            : auth_string_{username + ":" + password} {}
     Authentication(std::string&& username, std::string&& password)
-            : username_(std::move(username)),
-              password_(std::move(password)), auth_string_{this->username_ + ":" +
-                                                           this->password_} {}
+            : auth_string_{std::move(username)} {
```
Comment (COM8) [inline include/cpr/auth.h:15]: Any reason against:
```C++
Authentication(std::string&& username, std::string&& password)
    : auth_string_(std::move(username) + ":" + std::move(password)) {
}
```

Comment (jagerman) [reply include/cpr/auth.h]: No reason; updated.

**Comment by COM8** [comment]
Great work.Thanks. 👍🏻

**Pull request merged**